### PR TITLE
Missing escrow substore exists method

### LINKED ISSUE #8339: Missing escrowSubstoreExists methon on token module
### Description

Method `escrowSubstoreExists` was introduced in LIP PR https://github.com/LiskHQ/lips/pull/259

**Specification:** https://github.com/LiskHQ/lips/pull/259/files#diff-851f39ba2dc407b594e12baee157421d0efd44abd262b8d24cd1b7d20ae0dd8fR1554-R1558

### Acceptance Criteria

- Should have all unit tests passing

### Additional Information

Related LIP PR https://github.com/LiskHQ/lips/pull/259

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #8339

### How was it solved?

- [🌱 Add escrowSubstoreExists method](https://github.com/LiskHQ/lisk-sdk/commit/9842b84d267b52a428973264c7a2a0e78b89f7bb)

- [♻️ Add tokenMethod property on interoperability module](https://github.com/LiskHQ/lisk-sdk/commit/dc13a9dedd5091543e26afc58cfdfb0af3b7139f)

### How was it tested?

Run `npm run test:unit` under `framework`

### CODE REVIEW

File: framework/src/modules/token/method.ts [hunk 1]
```diff
@@ -708,4 +708,15 @@ export class TokenMethod extends BaseMethod {
...
+
+	public async escrowSubstoreExists(
+		methodContext: MethodContext,
+		chainID: Buffer,
+		tokenID: Buffer,
+	) {
+		const escrowStore = this.stores.get(EscrowStore);
+		const escrowExist = await escrowStore.has(methodContext, escrowStore.getKey(chainID, tokenID));
+
+		return escrowExist;
```
Comment (Incede) [inline framework/src/modules/token/method.ts:720]:
```suggestion
		return this.stores.get(EscrowStore).has(methodContext, escrowStore.getKey(chainID, tokenID));
```
probably better to be succinct similar to `isTokenSupported` method in the same file

File: framework/test/unit/modules/token/method.spec.ts [hunk 2]
```diff
@@ -1223,4 +1223,22 @@ describe('token module', () => {
 			});
 		});
 	});
+
+	describe('escrowSubstoreExists', () => {
+		const escrowChainID = defaultForeignTokenID.slice(0, CHAIN_ID_LENGTH);
+
+		it('should return false if escrow subStore does not exist', async () => {
```
Comment (Incede) [inline framework/test/unit/modules/token/method.spec.ts:1230]:
```suggestion
		it('should return false if escrow subStore does not exist for the given chain id and token id', async () => {
```
minor optional suggestion

File: framework/test/unit/modules/token/method.spec.ts [hunk 3]
```diff
@@ -1223,4 +1223,22 @@ describe('token module', () => {
...
+			const escrowStore = tokenModule.stores.get(EscrowStore);
+			await escrowStore.del(methodContext, escrowStore.getKey(escrowChainID, defaultTokenID));
+			await expect(
+				method.escrowSubstoreExists(methodContext, escrowChainID, defaultTokenID),
+			).resolves.toBeFalse();
+		});
+
+		it('should return true if escrow subStore exists', async () => {
```
Comment (Incede) [inline framework/test/unit/modules/token/method.spec.ts:1238]:
```suggestion
		it('should return true if escrow subStore exists for the given chain id and token id', async () => {
```
same

**Review by Incede — APPROVED** [review]
Looks good to me, I still made a few subjective optional suggestions!

**Review by gkoumout — APPROVED** [review]

**Pull request merged**